rspace/src/main/scala/coop/rchain/rspace/RSpaceOps.scala

```diff
@@ -140,16 +141,20 @@ abstract class RSpaceOps[F[_]: Concurrent, C, P, E, A, R, K](
     }
 
   override def reset(root: Blake2b256Hash): F[Unit] =
-    store.withTxnF(store.createTxnWriteF()) { txn =>
-      store.withTrieTxn(txn) { trieTxn =>
-        store.trieStore.validateAndPutRoot(trieTxn, store.trieBranch, root)
-        val leaves = store.trieStore.getLeaves(trieTxn, root)
-        eventLog.update(kp(Seq.empty))
-        store.getAndClearTrieUpdates()
-        store.clear(txn)
-        restoreInstalls(txn)
-        store.bulkInsert(txn, leaves.map { case Leaf(k, v) => (k, v) })
-      }
+    store.withTxnFlatF(store.createTxnWriteF()) { txn =>
+      for {
+        leaves <- store
+                   .withTrieTxn(txn) { trieTxn =>
+                     store.trieStore.validateAndPutRoot(trieTxn, store.trieBranch, root)
+                     store.trieStore.getLeaves(trieTxn, root)
+                   }
+                   .pure[F]
+        _ = eventLog.update(kp(Seq.empty))
+        _ = store.getAndClearTrieUpdates()
+        _ = store.clear(txn)
+        _ <- restoreInstalls(txn)
+        _ = store.bulkInsert(txn, leaves.map { case Leaf(k, v) => (k, v) })
```

Comment:
we shouldn't rely on the `_ =` hack because we lose guidance from the type system and might accidentaly `_ = x` where x is of type `F[A]` and thus lose effects.

I'd rather see a `.delay` in every line above, which has the additional value of screaming "please ~~kill~~ refactor me"